src/core/server/workspaces/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -138,6 +172,37 @@ export class WorkspaceSavedObjectsClientWrapper {
       return await wrapperOptions.client.delete(type, id, options);
     };
 
+    const updateWithWorkspacePermissionControl = async <T = unknown>(
+      type: string,
+      id: string,
+      attributes: Partial<T>,
+      options: SavedObjectsUpdateOptions = {}
+    ): Promise<SavedObjectsUpdateResponse<T>> => {
+      if (this.isRelatedToWorkspace(type)) {
+        await this.validateSingleWorkspacePermissions(id, wrapperOptions.request, [
+          PermissionMode.LibraryWrite,
+          PermissionMode.Management,
+        ]);
+      }
+      return await wrapperOptions.client.update(type, id, attributes, options);
+    };
+
+    const bulkUpdateWithWorkspacePermissionControl = async <T = unknown>(
+      objects: Array<SavedObjectsBulkUpdateObject<T>>,
+      options?: SavedObjectsBulkUpdateOptions
+    ): Promise<SavedObjectsBulkUpdateResponse<T>> => {
+      for (const object of objects) {
+        if (this.isRelatedToWorkspace(object.type)) {
+          await this.validateSingleWorkspacePermissions(object.id, wrapperOptions.request, [
+            PermissionMode.LibraryWrite,
+            PermissionMode.Management,
+          ]);
+        }
+      }
```

Comment:
There is a getPermittedWorkspaceIds method in permissionControlService, we'd better use that.

Reply:
And only management can update metadata.